Lib/locale.py

```diff
@@ -633,6 +633,7 @@ def getpreferredencoding(do_setlocale = True):
                 if not result and sys.platform == 'darwin':
```

Comment:
Why? This function would still have inconsistent behavior, if CODESET is set and if on windows it is still returning a value. Are suggesting that this function should not return anything?